src/sgl/func/base_module.cpp

```
@@ -0,0 +1,49 @@
+
+const std::string& BaseModule::name() const
+{
+    SGL_CHECK(m_module, "BaseModule has no Slang module");
+    return m_module->name();
+}
+
+void BaseModule::on_hot_reload(ref<SlangModule> module, ref<const sgl::ProgramLayout> low_level_layout)
+{
+    SGL_CHECK(module, "BaseModule hot reload requires a Slang module");
+    SGL_CHECK(low_level_layout, "BaseModule hot reload requires a low-level layout");
+    m_module = std::move(module);
```

Cppcheck (2.20.0)

[style] The function 'class_name' is never used.

(unusedFunction)